### PR TITLE
HZN-1566: Enforce VertexRef to be immutable 

### PR DESCRIPTION
**Pull request by patrick-schweizer**
* JIRA: https://issues.opennms.org/browse/HZN-1566

Setting to triage since the PR needs to be rebased to features/graph-service once https://github.com/OpenNMS/opennms/pull/2465 is merged.

Our [continuous integration system](http://bamboo.internal.opennms.com:8085) will test and verify your changes.

### CODE REVIEW

File: features/graph/api/src/main/java/org/opennms/netmgt/graph/api/Vertex.java [hunk 1]
```diff
@@ -30,7 +30,15 @@
...
 
-public interface Vertex extends VertexRef {
+public interface Vertex {
 
     GenericVertex asGenericVertex();
+
+    VertexRef getVertexRef();
+
+    // TODO: patrick discuss with mvr: shall we keep is method as a convenient way to say getVertexRef.getNamespace()
+    String getNamespace();
```
Comment (patrick-schweizer) [inline features/graph/api/src/main/java/org/opennms/netmgt/graph/api/Vertex.java]: @mvrueden to be discussed

Comment (mvrueden) [reply features/graph/api/src/main/java/org/opennms/netmgt/graph/api/Vertex.java]: Let's keep it for now. 
We can - if all features are implemented - still rework bits and pieces.

**Comment by mvrueden** [comment]
Will review this after HZN-1521 is merged. Please adjust commit messages, as they reference the wrong issue number.

**Comment by patrick-schweizer** [comment]
> Will review this after HZN-1521 is merged. Please adjust commit messages, as they reference the wrong issue number.

done. The relevant commits are:
https://github.com/OpenNMS/opennms/pull/2500/commits/d27fad2dd280286e02810119c68d80e9f0fda2ba and 
https://github.com/OpenNMS/opennms/pull/2500/commits/ff79a626d41514ce348f1468ceec0a1ec7d4d299

**Comment by patrick-schweizer** [comment]
removing triage label since I rebased the branch on the current features/grapgh-service branch

**Comment by mvrueden** [comment]
LGTM

**Pull request merged**